File: benchmarks/java/B6FileIO.java
import java.io.*;
import java.nio.file.*;
public class B6FileIO {
    public static void main(String[] args) throws Exception {
        int CHUNK = 1024 * 1024, CHUNKS = 4096;
        String fname = args.length > 0 ? args[0] : "../../data/fileio_test_java.tmp";
        byte[] buf = new byte[CHUNK];
        java.util.Arrays.fill(buf, (byte) 'A');
        FileOutputStream fos = new FileOutputStream(fname);
        try (OutputStream os = new BufferedOutputStream(fos)) {
            for (int i = 0; i < CHUNKS; i++) os.write(buf);
            os.flush();
            fos.getFD().sync();
        }
        long total = 0;
        try (InputStream is = new BufferedInputStream(new FileInputStream(fname))) {
            int n; while ((n = is.read(buf)) > 0) total += n;
        }
        System.out.println(total);
        Files.deleteIfExists(Path.of(fname));
    }
}

File: benchmarks/kotlin/b6_file_io.kt
import java.io.*
fun main(args: Array<String>) {
    val CHUNK = 1024 * 1024; val CHUNKS = 4096
    val fname = if (args.isNotEmpty()) args[0] else "../../data/fileio_test_kotlin.tmp"
    val buf = ByteArray(CHUNK) { 65 } // 'A'
    val fos = FileOutputStream(fname)
    BufferedOutputStream(fos).use { os ->
        repeat(CHUNKS) { os.write(buf) }
        os.flush()
        fos.fd.sync()
    }
    var total = 0L
    BufferedInputStream(FileInputStream(fname)).use { ins ->
        while (true) { val n = ins.read(buf); if (n < 0) break; total += n }
    }
    println(total)
    File(fname).delete()
}
